Merge dev into master (#130)
satisfactorymodding/Documentation: modules/ROOT/pages/Development/BeginnersGuide/dependencies.adoc

Be sure to read over the steps here as well to be sure you're doing things correctly.
====

== Before you Begin

Make sure you've read the starting information on the
xref:Development/BeginnersGuide/index.adoc[Getting Started title page]
to learn important information about how to use these docs.

== Satisfactory

This one should be pretty straightforward, and odds are, you've already done it.
Simply buy, download, and install Satisfactory from the Epic Games Launcher or Steam.
Launch the game at least once to ensure all files get set up correctly.

== Visual Studio

Visit the https://visualstudio.microsoft.com/downloads/[Microsoft Visual Studio downloads page]
and download the version you want.
We suggest either Visual Studio 2019 or 2017 Community,
because they're free of charge.

> It is important that if you use 2019 you still install the https://visualstudio.microsoft.com/vs/older-downloads/[2017 compilers].

Reviewer: Will wait to hear back before resolving this

> It is important that if you use 2019 you still install the https://visualstudio.microsoft.com/vs/older-downloads/[2017 compilers].

Author: Just checked and I indeed don't have 2017 build tools installed. Works just fine
Can't do a suggestion because it only selects the latest line but can I change the snippet to recommend VS22 and remove the info about the build tools?

> It is important that if you use 2019 you still install the https://visualstudio.microsoft.com/vs/older-downloads/[2017 compilers].

Reviewer: Go for it, yeah

> Visit the https://visualstudio.microsoft.com/downloads/[Microsoft Visual Studio downloads page]
> and download the version you want.
> We suggest either Visual Studio 2019 or 2017 Community,
> because they're free of charge.
> It is important that if you use 2019 you still install the https://visualstudio.microsoft.com/vs/older-downloads/[2017 compilers].

Author: Resolves earlier conversation on needing VS17 build tools
```suggestion
Visit the https://visualstudio.microsoft.com/downloads/[Microsoft Visual Studio downloads page]
and download the version you want.
We suggest either Visual Studio 2022 Community,
because it is free of charge and modern.
```

**** Visual Studio 2019

Click `Next`.
You don't need to add any plugins so just press `Install` to skip in the bottom left to begin the installation process.
Accept the terms and conditions prompts that appear along the way.

[TIP]
====
If you encounter a checksum error during Wwise integration,
installing any other Wwise version should fix the problem.
====

== Starter Project

The SML development team has an Unreal project prepared
that comes pre-installed with Satisfactory Mod Loader,
placeholder assets for base game content,
and some useful developer plugins.

Within this 'Starter Project,' you can develop as many mods as you like
while still keeping their files separate,
which will be covered later in the docs.

If you are familiar with version control software (ex. Git),
we suggest that you https://github.com/satisfactorymodding/SatisfactoryModLoader/[clone the repo],
which drastically simplifies the process of updating the files later when the game updates.
Check on the Discord to find out which branch to clone,
although it's most likely `master`.

If you're not familiar with version control,
https://www.gitkraken.com/learn/git/tutorials[now is a great time to learn]!

> However, if you'd just like to get started modding,

Reviewer: Feel free to resolve this unless you want to change the git wording here and/or push more tutorials

> If you're not familiar with version control,
> https://www.gitkraken.com/learn/git/tutorials[now is a great time to learn]!
> However, if you'd just like to get started modding,

Author: ```suggestion
If you're not familiar with version control,
https://www.gitkraken.com/learn/git/tutorials[now is a great time to learn]! Using Version Control Software (VCS) is a great skill to have for developers and will greatly help with updating your project in the future, as well as protect you from losing your mod sources if you upload them to a hosting website like GitHub.
However, if you'd just like to get started modding,
```

> If you're not familiar with version control,
> https://www.gitkraken.com/learn/git/tutorials[now is a great time to learn]!
> However, if you'd just like to get started modding,

Reviewer: ```suggestion
If you're not familiar with version control,
https://www.gitkraken.com/learn/git/tutorials[now is a great time to learn]!
Using Version Control Software (VCS) is a great skill to have for developers
and will greatly help with updating your project in the future,
as well as protect you from losing your mod sources if you upload them to a hosting website like GitHub.

However, if you'd just like to get started modding,
```

> However, if you'd just like to get started modding,

Reviewer: My suggested edit follows the line break patterns, no content difference
